src/main/scala/mesosphere/marathon/core/instance/Instance.scala

```diff
@@ -229,10 +230,37 @@ object Instance {
     }
   }
 
-  case class Id(idString: String) extends Ordered[Id] {
-    lazy val runSpecId: PathId = Id.runSpecId(idString)
-    lazy val executorIdString: String = Id.executorIdString(idString)
+  sealed trait Prefix {
+    val value: String
+    override def toString: String = value
+  }
+  case object PrefixInstance extends Prefix {
+    override val value = "instance-"
+  }
+  case object PrefixMarathon extends Prefix {
+    override val value = "marathon-"
+  }
+  object Prefix {
+    def fromString(prefix: String) = {
+      if (prefix == PrefixInstance.value) PrefixInstance
+      else PrefixMarathon
+    }
+  }
 
+  case class Id(val runSpecId: PathId, val prefix: Prefix, uuid: UUID) extends Ordered[Id] {
```

Comment:
Is it save to assume that uuid is always of type `UUID`? See my comment on the test fixes as well.